components/core/src/clp/SysFileReader.cpp

```diff
@@ -0,0 +1,62 @@
+#include "SysFileReader.hpp"
+
+#include <sys/stat.h>
+#include <unistd.h>
+
+#include <cstddef>
+#include <cstdio>
+#include <span>
+
+#include "ErrorCode.hpp"
+
+using std::span;
+
+namespace clp {
+auto SysFileReader::try_read(char* buf, size_t num_bytes_to_read, size_t& num_bytes_read)
+        -> ErrorCode {
+    if (nullptr == buf) {
+        return ErrorCode_BadParam;
+    }
+
+    num_bytes_read = 0;
+    span dst_view{buf, num_bytes_to_read};
+    while (true) {
+        auto const bytes_read = ::read(m_fd.get_raw_fd(), dst_view.data(), num_bytes_to_read);
+        if (0 == bytes_read) {
+            break;
+        }
+        if (bytes_read < 0) {
+            return ErrorCode_errno;
+        }
+        dst_view = dst_view.subspan(num_bytes_to_read);
+        num_bytes_read += bytes_read;
+        num_bytes_to_read -= bytes_read;
+        if (0 == num_bytes_to_read) {
+            return ErrorCode_Success;
+        }
+    }
+    if (0 == num_bytes_read) {
+        return ErrorCode_EndOfFile;
+    }
```

Comment:
```suggestion
    while (false == dst_view.empty()) {
        auto const bytes_read = ::read(m_fd.get_raw_fd(), dst_view.data(), dst_view.size());
        if (0 == bytes_read) {
            break;
        }
        if (bytes_read < 0) {
            return ErrorCode_errno;
        }
        num_bytes_read += bytes_read;
        dst_view = dst_view.subspan(bytes_read);
    }
    if (false == dst_view.empty()) {
        // We are supposed to read more bytes, but there's no more to read
        return ErrorCode_EndOfFile;
    }
```
Does it make sense to you?

Reply:
Right my bad. Updated the comment